src/chroma/distance.nim

```diff
@@ -0,0 +1,66 @@
+## Implementation of CIEDE2000 color difference formula
+## 
+## See: https://en.wikipedia.org/wiki/Color_difference#CIEDE2000
+## 
+## The implementation follows: http://www2.ece.rochester.edu/~gsharma/ciede2000/ciede2000noteCRNA.pdf
+
+import colortypes, transformations, math
+
+const twentyfiveToSeventh = (25^7).float32
+
+proc myAtan(x, y: float32): float32 =
+  if x == 0 and y == 0:
+    return 0
+  elif x >= 0:
+    arctan2(x, y).radToDeg
+  else:
+    arctan2(x, y).radToDeg + 360
+
+func deltaE00*(c1, c2: ColorLAB, k_L, k_C, k_H = 1.float32): float32 =
+  let
+    C1 = sqrt(c1.a^2 + c1.b^2)
+    C2 = sqrt(c2.a^2 + c2.b^2)
+    CM = 0.5*(C1 + C2)
+    CM7 = CM^7
+    G = 0.5*(1 - sqrt(CM7 / (CM7 + twentyfiveToSeventh)))
+    # aa1 is c1.a prime
+    aa1 = (1 + G)*c1.a
+    aa2 = (1 + G)*c2.a
+    CC1 = sqrt(aa1^2 + c1.b^2)
+    CC2 = sqrt(aa2^2 + c2.b^2)
+    h1 = myAtan(c1.b, aa1)
+    h2 = myAtan(c2.b, aa2)
+    deltaL = c2.l - c1.l
+    deltaCC = CC2 - CC1
+    deltah =
+      if CC1 == 0 or CC2 == 0:
+        0.float32
+      elif abs(h2 - h1) <= 180:
+        h2 - h1
+      elif h2 - h1 > 180:
+        h2 - h1 - 360
+      else:
+        h2 - h1 + 360
+    deltaHH = 2*sqrt(CC1*CC2)*sin(degToRad(0.5*deltah))
+    LM = 0.5*(c1.l + c2.l)
+    CCM = 0.5*(CC1 + CC2)
+    hM =
+      if CC1 == 0 or CC2 == 0:
+        h1 + h2
+      elif abs(h2 - h1) <= 180:
+        0.5*(h1 + h2)
+      elif h2 - h1 > 180:
+        0.5*(h1 + h2 + 360)
+      else:
+        0.5*(h1 + h2 - 360)
+    T = 1 - 0.17*cos(degToRad(hM - 30)) + 0.24*cos(degToRad(2*hM)) + 0.32*cos(degToRad(3*hM + 6)) - 0.20*cos(degToRad(4*hM - 63))
```

Comment:
There are some lines that are too long. `nimpretty` should fix it automatically.

Reply:
yep, this is done by nimpretty